src/MPDevice.cpp

```diff
@@ -1151,6 +1156,54 @@ void MPDevice::loadSingleNodeAndScan(AsyncJobs *jobs, const QByteArray &address,
     }));
 }
 
+void MPDevice::deleteDataNodes(QStringList nodeNames, MessageHandlerCb cb, Common::DataAddressType addrType)
+{
+    for (int i = 0; i < nodeNames.size(); ++i)
+    {
+        qDebug() << "Deleting file for " << nodeNames[i];
+        MPNode* parentPt = findNodeWithNameInList(Common::DATA_ADDR_IDX == addrType ? dataNodes : notesLoginNodes, nodeNames[i], true);
+
+        if(!parentPt)
+        {
+            exitMemMgmtMode(true);
+            qCritical() << "Couldn't find node for " << nodeNames[i];
+            cb(false, "Moolticute Internal Error (DDNAL#1)");
+        }
+        else
+        {
+            /* Delete all the childs */
+            deleteDataParentChilds(parentPt, addrType);
+
+            /* Delete empty parent
+             * We do not need CRED_ADDR_IDX, but cannot call without it
+             */
+
+            removeEmptyParentFromDB(parentPt, true, Common::CRED_ADDR_IDX, addrType);
+        }
+    }
+}
+
+bool MPDevice::checkDataNodeChanged(AsyncJobs *jobs, Common::DataAddressType addrType)
```

Comment:
as this function actually performs an action, could you rename it? maybe prepend "AndGeneratePacket"?